### ISSUE TITLE
Creating a flashable `-img.zip` file - experiment

### ISSUE THREAD

**Issue by petefoth** [lineageos4microg/docker-lineage-cicd#474]
We want to create a `-img.zip` file, containing all the `.img` files needed for each device. This file has two uses:
1. It is 'flashable' using `fastboot`, providing a simple way of installng the ROM without needing a custom recovery;
2. It contains all the `.img` files referenced in the LineageOS installation instructions. The list of needed `.img` files is device-specific, and is managed in the upstream device makefiles.

Currently we distribute these files by copying specific `.img` files (`recovery boot vendor_boot dtbo super_empty vbmeta`) from the`IMAGES` subdirectory of the `out/target/product/<device-name>/` directory tree. This is the current solution to #405 'Problems flashing devices which have their recovery in vendor_boot partition'.

A better solution is to use `mka dist` which, among many other outputs, creates a `lineage_$codename-img-eng.$USER.zip` in the `out/target/product/<device-name>/` directory, which we could then copy or move to the `zips` directory.

***Current situation***

This functionality is 'work in progress' in [the `pf-use-mka-dist` branch](https://github.com/lineageos4microg/docker-lineage-cicd/tree/pf-use-mka-dist): 
- `build.sh` [has been modified](https://github.com/lineageos4microg/docker-lineage-cicd/commit/0120c0eebe6f9c9e0620160c7d23e58452336fb0) to call `mka bacon && mka dist` instead of just `mka bacon`
- if the `INCLUDE_PROPRIETARY` environment variable is set to `false` this works
- if it is set to `true` then the operation fails (when the `mka dist` task is  about 85-95% complete, by which time the `-img.zip` file has been successfully created in the `out/target/product/<device-name>/` directory)  with the error `posix_spawn: Argument list too long`
- the `posix_spawn` error indicates that `mka` has initiated a command with a list of arguments that exceeds the maximum permitted size (`getconf ARG_MAX`)

***Possible fixes***
1. Use a different `mka` target to build the `-img.zip` file ***without*** the additional stuff which is causing `mka dist` to fail;
2. Find anther way of creating the  `-img.zip` file 
3. ??? Any more ideas welcome

**Comment by petefoth** [comment]
I'm closing this. The more I think about it, the more I am convinced that the only file we should be delivering is the ROM `.zip` file. Other  `.img` and `.zip` files are not necessary. I'll be making a new issue soon to deal with that

**Comment by petefoth** [comment]
The solution:
-------------
So, the answer is
1. make the target-files-package target 
`mka bacon target-files-package` 
  which  creates file `lineage_instantnoodle-target_files-eng.pete.zip` in `out/target/product/instantnoodle/obj/PACKAGING/target_files_intermediates/`
2. call build/tools/releasetools/img_from_target_files with that file as the input file to create the `-img.zip` file. 
```
build/tools/releasetools/img_from_target_files \ 
    out/target/product/instantnoodle/obj/PACKAGING/target_files_intermediates/lineage_instantnoodle-target_files-eng.pete.zip  \
    out/target/product/instantnoodle/lineage-20.0-20230907-instantnoodle-IMG.zip
```

The created -img.zip has all the .img files, and the android-info.txt, so it *should be* flashable, and it definitey contains the .img files we need to ship (in addition to the main ROM .zip file)
```
unzip -l out/target/product/instantnoodle/lineage-20.0-20230907-instantnoodle-img.zip

Archive:  out/target/product/instantnoodle/lineage-20.0-20230907-instantnoodle-img.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
       28  2008-01-01 00:00   android-info.txt
100663296  2008-01-01 00:00   boot.img
 25165824  2008-01-01 00:00   dtbo.img
292528128  2008-01-01 00:00   odm.img
473448788  2008-01-01 00:00   product.img
100663296  2008-01-01 00:00   recovery.img
     4952  2008-01-01 00:00   super_empty.img
886133172  2008-01-01 00:00   system.img
377286948  2008-01-01 00:00   system_ext.img
  1004220  2008-01-01 00:00   userdata.img
     8192  2008-01-01 00:00   vbmeta.img
     4096  2008-01-01 00:00   vbmeta_system.img
900427776  2008-01-01 00:00   vendor.img
---------                     -------
3157338716                     13 files
```

**Comment by petefoth** [comment]
To Do
------
- find out how to call `build/tools/releasetools/img_from_target_files` from  `src/build.sh`
- change  `src/build.sh`
    - update the `mka bacon` call
    - replace the current copying of `.img` files with a call to `build/tools/releasetools/img_from_target_files`
- test the changes

**Comment by vanMacG** [comment]
@petefoth Just a small note: The img.zip also contains the system imge, as does the main rom zip. So the needed disk space is doubled compared to today. I think we also had space problems on the download server before... Just keep it in mind ;)

**Comment by petefoth** [comment]
This is now implemented  in [the `pf-make-img-zip` branch](https://github.com/lineageos4microg/docker-lineage-cicd/tree/pf-make-img-zip), and the image is pushed to [dockerhub](https://hub.docker.com/layers/lineageos4microg/docker-lineage-cicd/pf-make-img-zip/images/sha256-b842cb1eb5fcd0efcfbfd878e531aba2450f881d7676d0eb7e6b96051638d2d4?context=explore) (pull command `docker pull lineageos4microg/docker-lineage-cicd:pf-make-img-zip`)

I have used it to make builds for `lilac`, `bluejay` & `instantnoodle`. In each case the `-img.zip` file contains the `.img` files needed by the LOS installation instructions. And - the `lilac` file at least - is also flashable using `fastboot`. But, as @vanMacG poinnted out, they also contain other large `.img`s (`vendor`, `system`, `system_ext`) so the `-img.zip` files are larger than the main ROM `.zip` file. As we do not have a lot of spare space on the download server (25GB free out of 590GB total) this is not a sensible way of delivering the needed `.img` files, so I won;t be taking it any further. I earned a lot doing it though :)
